src/groupeddatatable/grouping.jl

```diff
@@ -236,12 +236,28 @@ colwise(sum, groupby(dt, :a))
 ```
 
 """
-colwise(f::Function, d::AbstractDataTable) = map(i -> f(d[i]), 1:ncol(d))
+function colwise(f::Function, d::AbstractDataTable)
+    x = [f(d[i]) for i in 1:ncol(d)]
+    if eltype(x) <: Nullable && isa(x, Vector{eltype(x)})
+        return NullableArray(x)
+    end
+    return x
+end
 colwise(f::Function, gd::GroupedDataTable) = map(colwise(f), gd)
 colwise(f::Function) = x -> colwise(f, x)
 colwise(f) = x -> colwise(f, x)
 # apply several functions to each column in a DataTable
-colwise{T<:Function}(fns::Vector{T}, d::AbstractDataTable) = map(i -> map(f -> f(d[i]), fns), 1:ncol(d))
+function colwise{T<:Function}(fns::Vector{T}, d::AbstractDataTable)
+    res = Array{AbstractVector}(ncol(d))
```

Comment:
I don't have any strong feelings one way or the other. The same result can be achieved by `hcat(output...)`, but if the `Matrix` output is more useful and flexible then we might as well return that one by default.
```julia
julia> dt = DataTable(A = 1:10, B = 11:20)
10×2 DataTables.DataTable
│ Row │ A  │ B  │
├─────┼────┼────┤
│ 1   │ 1  │ 11 │
│ 2   │ 2  │ 12 │
│ 3   │ 3  │ 13 │
│ 4   │ 4  │ 14 │
│ 5   │ 5  │ 15 │
│ 6   │ 6  │ 16 │
│ 7   │ 7  │ 17 │
│ 8   │ 8  │ 18 │
│ 9   │ 9  │ 19 │
│ 10  │ 10 │ 20 │

julia> current = colwise([length, sum], dt)
2-element Array{AbstractArray{T,1},1}:
 Any[10,55]
 Any[10,155]

julia> hcat(current...)
2×2 Array{Any,2}:
 10    10
   55    155

julia> alternate = [f(dt[i]) for f in [length, sum], i in 1:ncol(dt)]
2×2 Array{Any,2}:
 10    10
   55    155
```